include/seastar/coroutine/exception.hh

```diff
@@ -30,7 +30,7 @@ namespace internal {
 struct exception_awaiter {
     std::exception_ptr eptr;
 
-    explicit exception_awaiter(std::exception_ptr&& eptr) : eptr(std::move(eptr)) {}
+    explicit exception_awaiter(std::exception_ptr&& eptr) noexcept : eptr(std::move(eptr)) {}
```

Comment:
Nitpick: although I am guessing exception_ptr is indeed, like you said, nothrow movable in practice, I am not sure this is actually mandated by the standard. Why was it important to add this "noexcept" here? Is it related to the purpose of this series?

Reply:
Not specifically for the purpose of this series, but in general, with `co_await coroutine::return_exception(some_exception())` you don't want an exception to escape from this "control" statement (although it is not a bug).  This behavior is akin to future exception and continuation handling where we terminate if we fail to create a continuation or make an exceptional future.